### DISCUSSION TITLE
How to compute which DataGrid column was clicked

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#15583] [Q&A]
I have a DataGrid class derived from Avalonia.Controls.DataGrid.

I added a handler for PointerPressedEvent.

Is there an easy way to compute which column was clicked, accounting for complications like scroll state etc.?

**Comment by timunie** [comment]
e.Source should give you the control that was clicked. Use VisualTreeExtensions to find the first parent of type DataGridCell. This control has info about it's row and col iirc.
